allennlp/modules/vision/proposal_generator.py

```diff
@@ -17,9 +28,300 @@ def forward(self, images: FloatTensor):
         raise NotImplementedError()
 
 
-@ProposalGenerator.register("null")
-class NullProposalGenerator(ProposalGenerator):
-    """A `ProposalGenerator` that never returns any proposals."""
+# A `ProposalGenerator` takes a batch of images as a tensor with the dimensions
+# (Batch, Color, Height, Width), and returns bounding box information below:
+#     1) box coordinates(Batch,  # Boxes, 4)
+#     2) box classification probability(Batch,  # Boxes, #classes)
+#     3) ROI - pooled box features(Batch,  # Boxes, #features)
+
+from allennlp.common.file_utils import cached_path
+import tarfile
+from filelock import FileLock
+import shutil
+import os
+
+def _load_detectron_checkpoint(checkpoint_name="RCNN-X152-C4-2020-07-18"):
+    """
+    Download and cache a tarfile containing detectron checkpoint (_weights.th)
+    It might be better to replace this with native utility functions like load_archive (https://github.com/allenai/allennlp/blob/master/allennlp/models/archival.py#L132) later,
+    where the tarfile includes 1) config.json 2) _weights.th
+    """
+    gs_url = "https://storage.googleapis.com/allennlp-public-models"
+    detectron_checkpoint_url = f'{gs_url}/{checkpoint_name}.tar.gz'
+    extraction_path = cached_path(detectron_checkpoint_url, extract_archive=True)
+
+    folder_name = detectron_checkpoint_url.split('/')[-1].split('.')[0]  # ex) RCNN-X152-C4-2020-07-18
+    weights_path = f'{extraction_path}/{folder_name}/_weights.th'
```

Comment:
You can get the path to a file inside of an extracted archive from `cached_path()`. For example:
```Python
cached_path("http://foo.com/bar.tar.gz!file/inside/archive.txt", extract_archive=True)
```

Reply:
Oh I didn't know that.
This looks much better. I'll re-write in this way.